client/session.go

```diff
@@ -1521,6 +1521,10 @@ func (s *session) readConsistencyResult(
 	return nil
 }
 
+func (s *session) IteratorPools() (encoding.IteratorPools, error) {
+	return s.pools, nil
```

Comment:
Might want to check that `s.pools != nil`?  Also are the pools always guaranteed to be available (I can't remember when they get alloc'd).  If not we probably have to nil check the specific pools in question too and return an error if nil perhaps?